dissect/target/plugins/apps/av/sophos.py

```diff
@@ -98,22 +99,30 @@ def sophoshomelogs(self) -> Iterator[SophosLogRecord]:
             path (path): Path to the infected file (if available).
 
         """
-        log = self.target.fs.path(self.LOG_SOPHOS_HOME).open("rt", 0, "utf-16le")
-        while line := log.readline():
-            if line.find(self.MARKER_INFECTION) > -1:
-                try:
-                    ts, json_data = line.split(" ", maxsplit=2)
-                    details = json.loads(json_data)
-
-                    path_to_infected_file = None
-                    if targets := details.get("targets", None):
-                        path_to_infected_file = targets[0].get("file_path", None)
-
-                    yield SophosLogRecord(
-                        ts=ts,
-                        description=details.get("threat_name", details),
-                        path=self.target.fs.path(path_to_infected_file),
-                        _target=self.target,
-                    )
-                except Exception as error:
-                    self.target.log.warning(f"Error: {error} on log line: {line}.")
+        if self.target.fs.path(self.LOG_SOPHOS_HOME).exists():
+            for line in self.target.fs.path(self.LOG_SOPHOS_HOME).open("rt", 0, "utf-16le"):
+                if line.find(self.MARKER_INFECTION) > -1:
+                    try:
+                        ts, json_data = line.split(" ", maxsplit=2)
+                        details = json.loads(json_data)
+
+                        path_to_infected_file = None
+                        if targets := details.get("targets", None):
+                            path_to_infected_file = targets[0].get("file_path", None)
+
+                        # < 3.9.4.1
+                        if path := details.get("file_path"):
+                            path_to_infected_file = path
+                        # > 3.10.3
+                        elif targets := details.get("targets", None):
+                            path_to_infected_file = targets[0].get("file_path", None)
+                            path = (self.target.fs.path(path_to_infected_file),)
+
+                        yield SophosLogRecord(
+                            ts=ts,
+                            description=details.get("threat_name", details),
+                            path=self.target.fs.path(path_to_infected_file),
+                            _target=self.target,
+                        )
+                    except Exception as error:
+                        self.target.log.warning(f"Error: {error} on log line: {line}.")
```

Comment:
```suggestion
                        self.target.log.warning("Error: {error} on log line: {line}.")
```

Reply:
Make it a proper logging format string using `%s` and passing `error` as argument.